### PR TITLE
Add extent shards for ScenarionMetadata and ScenarioRunMetadata

### PR DESCRIPTION
**Pull request by mahkoCosmo**

### CODE REVIEW

File: src/main/kotlin/com/cosmotech/api/azure/eventhubs/AzureEventHubsClient.kt [hunk 1]
```diff
@@ -133,6 +135,8 @@ class AzureEventHubsClient(private val csmPlatformProperties: CsmPlatformPropert
         "${scenarioRunMetaData.simulationRun},${scenarioRunMetaData.scenarioId}," +
             "${scenarioRunMetaData.scenarioRunStartTime}"
     val eventData = EventData(data)
+    val dropTag = "drop-by:" + scenarioRunMetaData.scenarioId
```
Comment (vcarluer) [inline src/main/kotlin/com/cosmotech/api/azure/eventhubs/AzureEventHubsClient.kt:138]: the drop-by key here should be the csmSimulationId (scenarioRunMetaData.simulationRun ?) to be able to delete 1 scenario run and have the related data deleted: probesmeasures, simulationtotalfacts, scenariorunmetadata.
In current implementation all scenarunrunmetadata data will be be deleted for the scenario and not the data for only 1 scenario run.

**Review by vcarluer — CHANGES_REQUESTED** [review]
Reviewed with @jreynard-code

File: src/main/kotlin/com/cosmotech/api/azure/eventhubs/AzureEventHubsClient.kt (hunk 1 above)
Comment (mahkoCosmo) [reply src/main/kotlin/com/cosmotech/api/azure/eventhubs/AzureEventHubsClient.kt:138]: In the case that there no simulationrun, when the user delete a scenario we have to delete all scenario metadata that why the drop-by key has to be 'scenarioId'

Comment (vcarluer) [reply src/main/kotlin/com/cosmotech/api/azure/eventhubs/AzureEventHubsClient.kt:138]: I do not agree: 
* No scenario run = no need to delete from scenariorunmetadata
* csmSimulationRun is a SimulationRun property
* scenariometadata should be deleted in the ScenarioServiceImpl.deleteScenario endpoint not scenariorun

**Review by vcarluer — APPROVED** [review]

**Pull request merged**